jenkins-scm-koji-plugin/src/main/java/hudson/plugins/scm/koji/client/KojiBuildMatcher.java

```diff
@@ -139,16 +139,34 @@ private boolean matchesTagPredicate(Set<String> tags) {
      * here so we can store it.
      */
     private List<RPM> retrieveArchives(Build build) {
-        final List<String> supportedArches = new ArrayList<>(1);
-        supportedArches.add("win");
         final XmlRpcRequestParams params = new ListArchives(build.getId(), null);
         final ArchiveList response = ArchiveList.create(execute(build.getProvider().getTopUrl(), params));
         final List<String> archivefilenames = response.getValue();
         if (archivefilenames == null || archivefilenames.isEmpty()) {
             return Collections.emptyList();
         }
         final List<RPM> archives = new ArrayList<>(archivefilenames.size());
+        if (isBuildContainer(build)) {
+            //for contaiers, we use different approach(because they havee arch in name only), we list all the archives, and include only those of arch (all if no arch is specified)
```

Comment:
Please, extract this as a method incl.comment.

Reply:
Wrong selection, I meant this:

//for contaiers, we use different approach(because they havee arch in name only), we list all the archives, and include only those of arch (all if no arch is specified)
             //luckily for us, they are mess.arch.tar.gz
            if (archs == null || archs.isEmpty()) {
                addAllContainerArchives(archivefilenames, archives, build);
            } else {
                addSelectedArchesContainers(archivefilenames, archives, build);
            }

Reply:
Do you really insists? That cmment is for both if/else  parts of this if.

Reply:
From my point of view is comment useless. It's written in code and it's clear from if-else that we need to use different approaches.